### ISSUE TITLE
towns not detected correctly in reverse search

### ISSUE THREAD

**Issue by simonm3** [openstreetmap/Nominatim#1597]
I am looking at villages in Nigeria that are clearly different settlements, 15km apart and marked on osm with village name. These do not produce correct results on reverse geocode. Possiby because they are nodes. However Photon, which is also OSM based reports them correctly.

These two are reported as  | Kadadi, Gada, Sokoto, Nigeria. 
Photon reports them correctly

> 13.862500 | 5.558333 | (Rafinduma, Sokoto, Nigeria
> 13.845834 | 5.587499 | (Gidan Tanko, Sokoto, Nigeria

This one in same area is reported as Rungumawa Gatti, Illela, Sokoto, Nigeria
Photon reports it correctly as 
> 13.832084 | 5.447499 | (Tupal, Sokoto, Nigeria

**Comment by mtmail** [comment]
Nominatim needs to do some guessing finding the nearest town or village. That's usually by distance. It seems for these searches those villages mapped as relations get preferred.

You can see the difference between search position (red small circle) and result (blue larger circle) on https://nominatim.openstreetmap.org/reverse.php?format=html&lat=13.862500&lon=5.558333
and the address components when clicking 'details' on a result
https://nominatim.openstreetmap.org/details.php?osmtype=W&osmid=532970405
Here some villages are grayed-out, that's candidates that haven't been chosen.

Not sure yet how easy it is to fix that, without breaking something else.

(Distances not displayed on the page seems an unrelated issue, I see them fine in the raw JSON output https://nominatim.openstreetmap.org/details.php?osmtype=W&osmid=532970405&class=highway&format=json&addressdetails=1)

**Comment by simonm3** [comment]
Why are there some names that are relations and others as nodes? People live in a village that may be part of an area which is part of a country. One might expect that to be presented as village, area country. If there is a limit of only 4 levels in the hierarchy then these villages should be separate places/relations because they are clearly distinct. It is a data issue.

I ddnt meant the distances are not displayed just that the villages are 15km apart. There is no ambiguity. They are distinct places.

**Comment by mtmail** [comment]
Having several villiages (tagged admin_level=8, nodes) inside an villiage (tagged admin_level=8, relation) makes it hard to impossible for Nominatim to make a good judgement. I can't tell if the relation should have a lower admin_level or the villages a higher one. Or if the relation should instead be a node, the mapper who created it might know. As you say https://wiki.openstreetmap.org/wiki/Tag:boundary%3Dadministrative only lists 4 levels for Nigeria.

Closing here as data issue.

Distances: I fixed the output in https://github.com/openstreetmap/Nominatim/pull/1601 It's not live yet.
